### ISSUE TITLE
typing.Annotated should be considered a dangerous default

### ISSUE THREAD

**Issue by lattwood** [pylint-dev/pylint#10093]
### Current problem

If a programmer accidentally uses `typing.Annotated` as a default value (*not* as a default type), pylint doesn't complain.

Ex, you're converting some FastAPI code from plain Depends to Annotated and make a typo, like below:

Original:
```
def some_method(
    session: AsyncSession = Depends(get_session_read),
) -> None:
    ...
```

Brainfart:
```
def some_method(
    session= typing.Annotated[AsyncSession, Depends(get_session_read)],
) -> None:
    ...
```

What you're actually supposed to do:
```
def some_method(
    session: typing.Annotated[AsyncSession, Depends(get_session_read)],
) -> None:
    ...
```

### Desired solution

PyLint should make me feel bad about myself for making this typo. /s

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. I'm not sure we can easily create a checker for this as it's possible to use typing classes as default value for function argument voluntarily (meta-programming or advanced use cases?).

I.e. something like this:
```python
def some_meta_method(
    typing_type: Type[typing.Annotated] = typing.Annotated[AsyncSession, Depends(get_session_read)],
) -> None:
    ...
```
How would you deal with that ? Raise only if there's no defined type for ``typing_type`` ?

**Comment by lattwood** [comment]
yeah, I think that would have to be the approach.
